src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -279,7 +334,7 @@ def __getitem__(self, key: slice) -> ArrayLike:
         # noqa: DAR101 key
         """
 
-    def __getitem__(self, key: SupportsIndex | datetime | slice) -> float | ArrayLike:
+    def __getitem__(self, key: SupportsIndex | datetime | slice) -> float:
```

Comment:
Shouldn't `datetime` and `slice` be removed from here?

Reply:
only slice should be removed, datetime is fine

Reply:
I thought mypy complained about using non-ints as indexes, but maybe I'm mixing up things...

Reply:
are damn! of course you are right, that was one of the motivations for this change.